core/services/ocr2/plugins/functions/config/config.go

```diff
@@ -1,23 +1,32 @@
 package config
 
 import (
+	"errors"
+	"fmt"
+
 	"google.golang.org/protobuf/proto"
 
+	decryptionPluginConfig "github.com/smartcontractkit/tdh2/go/ocr2/decryptionplugin/config"
+
 	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
 )
 
 // This config is part of the job spec and is loaded only once on node boot/job creation.
 type PluginConfig struct {
-	MinIncomingConfirmations        uint32                     `json:"minIncomingConfirmations"`
-	RequestTimeoutSec               uint32                     `json:"requestTimeoutSec"`
-	RequestTimeoutCheckFrequencySec uint32                     `json:"requestTimeoutCheckFrequencySec"`
-	RequestTimeoutBatchLookupSize   uint32                     `json:"requestTimeoutBatchLookupSize"`
-	PruneMaxStoredRequests          uint32                     `json:"pruneMaxStoredRequests"`
-	PruneCheckFrequencySec          uint32                     `json:"pruneCheckFrequencySec"`
-	PruneBatchSize                  uint32                     `json:"pruneBatchSize"`
-	ListenerEventHandlerTimeoutSec  uint32                     `json:"listenerEventHandlerTimeoutSec"`
-	MaxRequestSizeBytes             uint32                     `json:"maxRequestSizeBytes"`
-	GatewayConnectorConfig          *connector.ConnectorConfig `json:"gatewayConnectorConfig"`
+	MinIncomingConfirmations           uint32                     `json:"minIncomingConfirmations"`
+	RequestTimeoutSec                  uint32                     `json:"requestTimeoutSec"`
+	RequestTimeoutCheckFrequencySec    uint32                     `json:"requestTimeoutCheckFrequencySec"`
+	RequestTimeoutBatchLookupSize      uint32                     `json:"requestTimeoutBatchLookupSize"`
+	PruneMaxStoredRequests             uint32                     `json:"pruneMaxStoredRequests"`
+	PruneCheckFrequencySec             uint32                     `json:"pruneCheckFrequencySec"`
+	PruneBatchSize                     uint32                     `json:"pruneBatchSize"`
+	ListenerEventHandlerTimeoutSec     uint32                     `json:"listenerEventHandlerTimeoutSec"`
+	MaxRequestSizeBytes                uint32                     `json:"maxRequestSizeBytes"`
+	GatewayConnectorConfig             *connector.ConnectorConfig `json:"gatewayConnectorConfig"`
+	MaxDecryptionQueueLength           uint32                     `json:"maxDecryptionQueueLength"`
```

Comment:
Put in a sub-struct please. Otherwise it's too confusing.
I see you already have a struct in plugin.go, just put it here.

Reply:
My goal was to avoid having to nest these config values under another field in the jobspec TOML.  However, to make the code more readable I used embedding.